app/javascript/components/UserAssignmentForm.js

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import UserAssignmentFormField from "./UserAssignmentFormField";
+
+class UserAssignmentForm extends React.Component {
+  constructor(props) {
+    super();
+    this.state = props;
+    this.handleAddClick = this.handleAddClick.bind(this);
+    this.handleDeleteClick = this.handleDeleteClick.bind(this);
+  }
+
+  randomUUID() {
+    return "new-mission-"+ Math.floor(Math.random() * Math.floor(9000));
+  }
+
+  handleAddClick() {
+    let assignments = this.state.assignments;
+    assignments.push({role: "", mission: this.randomUUID(), new_assignment: true});
+    this.setState({assignments: assignments});
+  }
+
+  handleDeleteClick(mission, e) {
+    let assignments = this.state.assignments;
+    let idx = assignments.findIndex((a) => { return a.mission_id == mission; });
+    assignments[idx].new ? assignments.splice(idx, 1) : assignments[idx]["destroy"] = true
```

Comment:
Note that `obj["foo"]` is same as `obj.foo`. Please simplify accordingly.

Also please don't do assignment inside a ternary operator. Please use if/else.

Also same goes here for how to properly mutate an array in state. Can you please lookup how to safely remove an item?